### PR TITLE
Update DG

### LINKED ISSUE #196: [DOC] Radian Write at least 3 pages worth of DG in total

### LINKED ISSUE #205: [DOC] Clean Up Design and Implementation Artefact from AB3 DG Logic + Intro

### PR DESCRIPTION
**Pull request by radiankrisno**
### Summary

Resolves #196 
Resolves #205

### Detail

* Update Logic puml and developer guide
* Add implementation details for viewing patients
* Add some use cases

### Checks

- [X] Link PR to issue
- [X] Add Reviewer
- [X] Pass CI

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -308,6 +319,27 @@ As seen in the diagram above, after the `LogicManager` receives the `ProfileComm
...
+
+This feature can be accessed using the `view` command, which currently only support the viewing of patient which follows
+the following format `view t/patient PATIENT_ID` where `PATIENT_ID` is a valid id of a patient.
+
+Given below is the sequence diagram when a user provides an example of a valid `view` command
+(`view t/patient P001`) to see the complete details of the patient.
+
+![ViewPatientCommand](images/ViewSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `ViewPatientParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
```
Comment (simonjulianl) [inline docs/DeveloperGuide.md:333]: I think you don't need to repeat this over, I think I saw this somewhere else

**Review by simonjulianl — CHANGES_REQUESTED** [review]
Some redundant stuff

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -308,6 +319,27 @@ As seen in the diagram above, after the `LogicManager` receives the `ProfileComm
...
+</div>
+
+As seen in the diagram above, when `view` command is made by the user, following steps happened inside the logic:
+1. When `Logic` is called upon to execute a command, it uses the `AddressBookParser` class to parse the user command.
```
Comment (simonjulianl) [inline docs/DeveloperGuide.md:337]: All of these actually are pretty redundant, given you have sequence diagram. You can use activity diagram here too btw

**Review by simonjulianl — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (radiankrisno) [reply docs/DeveloperGuide.md]: got it

**Pull request merged**